### PR TITLE
fix(Curriculum): ensure code that includes and is not equal to the fl…

### LINKED ISSUE #38491: Intermediate Algorithm Scripting: false negative final test
**Describe your problem and how to reproduce it:**
The final test:

> Your solution should not use the Array.prototype.flat() or Array.prototype.flatMap() methods.

fails on code that does not use either of those methods.
For example,
```
function steamrollArray(arr) {
  let results = []
  function flatten(item) {
    if(!Array.isArray(item)){
      results.push(item)
    } else {
      item.forEach(flatten)
    }
  }
  arr.forEach(flatten)
  return results
}
```
passes all the tests, except for the last one, despite not using Array.prototype.flat() or Array.prototype.flatMap() while 
```
function steamrollArray(arr) {
  let results = []
  function squish(item) {
    if(!Array.isArray(item)){
      results.push(item)
    } else {
      item.forEach(squish)
    }
  }
  arr.forEach(squish)
  return results
}
```
successfully passes all the tests, even though the only difference is that the second example uses `squish` instead of `flatten`

Additionally, all of [FCC's official solutions for this challenge](https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-steamroller/16079) fail the last test except for solution 3, which is the only solution that does not contain 'flat' anywhere in the code

**Add a Link to the page with the problem:**
[steamroller challenge](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller)

**Tell us about your browser and operating system:**
* Browser Name: Chrome
* Browser Version: 80.0.3987.149
* Operating System: macOS 10.15.3 

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
Screen Shot 2020-04-02 at 1 13 18 PM

### PR DESCRIPTION
**Pull request by AshNaz87**
…at or flatMap methods do not fail the test suite

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38491

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller.english.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ tests:
   - text: <code>steamrollArray([1, {}, [3, [[4]]]])</code> should return <code>[1, {}, 3, 4]</code>.
     testString: assert.deepEqual(steamrollArray([1, {}, [3, [[4]]]]), [1, {}, 3, 4]);
   - text: Your solution should not use the <code>Array.prototype.flat()</code> or <code>Array.prototype.flatMap()</code> methods.
-    testString: assert(!code.match(/\.?[\s\S]*?flat/) && !code.match(/\.?[\s\S]*?flatMap/));
+    testString: assert(!code.match(/\.?[\s\S]*?flat$/) && !code.match(/\.?[\s\S]*?flatMap$/));
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller.english.md:36]: I am not sure why the dot `.` is before the `[\s\S]*?`. The `\S` should not be there.  I suggest:
```suggestion
    testString: assert(!code.match(/\w+?\s*\.flat\s*\(\d*\)/) && !code.match(/\w+?\s*\.flatMap\s*?\(/));
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@AshNaz87 Take a look at my suggestion and let me know what you think.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller.english.md (hunk 1 above)
Comment (AshNaz87) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller.english.md:36]: Makes sense. Didn't want to change anything unnecessary

**Comment by RandellDawson** [comment]
Why did you close this PR?

**Comment by AshNaz87** [comment]
@RandellDawson iirc, there were problems with the regex, so I thought it'd be best to start afresh

**Pull request closed without merging**